homeassistant/components/ukraine_alarm/config_flow.py

```diff
@@ -0,0 +1,157 @@
+"""Config flow for Ukraine Alarm."""
+import aiohttp
+from ukrainealarm.client import Client
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_API_KEY, CONF_NAME, CONF_REGION
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+
+
+class UkraineAlarmConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow for Ukraine Alarm."""
+
+    VERSION = 1
+
+    def __init__(self):
+        """Initialize a new UkraineAlarmConfigFlow."""
+        self.api_key = None
+        self.states = None
+        self.selected_region = None
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+
+        if user_input is not None:
+            websession = async_get_clientsession(self.hass)
+            try:
+                regions = await Client(
+                    websession, user_input[CONF_API_KEY]
+                ).get_regions()
+                if not regions:
+                    errors["base"] = "unknown"
+            except aiohttp.ClientResponseError as ex:
+                errors["base"] = "invalid_api_key" if ex.status == 401 else "unknown"
+            except aiohttp.ClientConnectionError:
+                errors["base"] = "cannot_connect"
+            except aiohttp.ClientError:
+                errors["base"] = "unknown"
+            else:
```

Comment:
Probably check if `errors` is falsy instead.

Reply:
Updated.